automatic/prtgdesktop/tools/chocolateyuninstall.ps1

```diff
@@ -0,0 +1,36 @@
+﻿$ErrorActionPreference = 'Stop';
+
+$packageArgs = @{
+  packageName   = $env:ChocolateyPackageName
+  softwareName  = 'PRTG Desktop'
+  fileType      = 'exe'
+  silentArgs    = "/quiet /norestart"
+  validExitCodes= @(0, 3010, 1605, 1614, 1641)
+}
+
+$uninstalled = $false
+
+[array]$key = Get-UninstallRegistryKey -SoftwareName $packageArgs['softwareName']
+
+if ($key.Count -eq 1) {
+  $key | % { 
+    $packageArgs['file'] = "$($_.UninstallString)"
+    
+    if ($packageArgs['fileType'] -eq 'MSI') {
+      $packageArgs['silentArgs'] = "$($_.PSChildName) $($packageArgs['silentArgs'])"
+
+      $packageArgs['file'] = ''
+    } else {
+
+    }
+
+    Uninstall-ChocolateyPackage @packageArgs
```

Comment:
Have you tested if an uninstall script is really necessary?

Chocolatey has an auto uninstall feature. If this work an uninstall script is not required. See [§1.1.11](https://github.com/chocolatey-community/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md#1111-provide-uninstaller-only-if-needed)

Reply:
Chocolatey auto uninstall works.